lightningd/options.c

```diff
@@ -860,6 +860,8 @@ static void handle_minimal_config_opts(struct lightningd *ld,
 			       &ld->config_dir,
 			       "Set working directory. All other files are relative to this");
 
+	ld->config_dir = path_join(ld, path_cwd(tmpctx), ld->config_dir);
```

Comment:
This leaks the old value, but the fix is simple:

`ld->config_dir = path_join(ld, path_cwd(tmpctx), take(ld->config_dir));`

FYI:
E            - Node /tmp/ltests-2io5d811/test_closing_while_disconnected_1/lightning-1/ has memory leaks: [
E               {
E                   "backtrace": [
E                       "ccan/ccan/tal/tal.c:437 (tal_alloc_)",
E                       "ccan/ccan/tal/tal.c:466 (tal_alloc_arr_)",
E                       "ccan/ccan/tal/tal.c:794 (tal_dup_)",
E                       "ccan/ccan/tal/str/str.c:18 (tal_strdup_)",
E                       "common/configdir.c:11 (opt_set_talstr)",
E                       "ccan/ccan/opt/parse.c:121 (parse_one)",
E                       "ccan/ccan/opt/opt.c:230 (early_parse)",
E                       "ccan/ccan/opt/opt.c:247 (opt_early_parse_incomplete)",
E                       "lightningd/options.c:841 (handle_minimal_config_opts)",
E                       "lightningd/options.c:1097 (handle_early_opts)",
E                       "lightningd/lightningd.c:681 (main)"
E                   ],
E                   "label": "common/configdir.c:11:char[]",
E                   "parents": [],
E                   "value": "0x224e5a8"

Reply:
Thanks, I just removed the condition and didn't check before push.. :/